pkg/blockio/blockio.go

```diff
@@ -0,0 +1,456 @@
+//       return err
+//   }
+//   // Output option 1: write directly to cgroup "/mytestgroup"
+//   if err := blockio.SetCgroupClass("/mytestgroup", "LowPrioThrottled"); err != nil {
+//       return err
+//   }
+//   // Output option 2: OCI LinuxBlockIO of a blockio class
+//   if lbio, err := blockio.OciLinuxBlockIO("LowPrioThrottled"); err != nil {
+//       return err
+//   } else {
+//       fmt.Printf("OCI LinuxBlockIO for LowPrioThrottled:\n%+v\n", lbio)
+//   }
+package blockio
+
+import (
+	"fmt"
+	"io/ioutil"
+	stdlog "log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"sigs.k8s.io/yaml"
+
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/intel/goresctrl/pkg/cgroups"
+	grclog "github.com/intel/goresctrl/pkg/log"
+)
+
+const (
+	// sysfsBlockDeviceIOSchedulerPaths expands (with glob) to block device scheduler files.
+	// If modified, check how to parse device node from expanded paths.
+	sysfsBlockDeviceIOSchedulerPaths = "/sys/block/*/queue/scheduler"
+)
+
+// tBlockDeviceInfo holds information on a block device to be configured.
+// As users can specify block devices using wildcards ("/dev/disk/by-id/*SSD*")
+// tBlockDeviceInfo.Origin is maintained for traceability: why this
+// block device is included in configuration.
+// tBlockDeviceInfo.DevNode contains resolved device node, like "/dev/sda".
+type tBlockDeviceInfo struct {
+	Major   int64
+	Minor   int64
+	DevNode string
+	Origin  string
+}
+
+// Our logger instance.
+var log grclog.Logger = grclog.NewLoggerWrapper(stdlog.New(os.Stderr, "[ blockio ] ", 0))
+
+// classBlockIO connects user-defined block I/O classes to
+// corresponding cgroups blockio controller parameters.
+var classBlockIO = map[string]cgroups.BlockIOParameters{}
+
+// SetLogger sets the logger instance to be used by the package.
+// Examples:
+//   // Log to standard logger:
+//   stdlog := log.New(os.Stderr, "blockio:", 0)
+//   blockio.SetLogger(goresctrllog.NewLoggerWrapper(stdlog))
+//   // Log to logrus:
+//   blockio.SetLogger(logrus.New())
+func SetLogger(l grclog.Logger) {
+	log = l
+}
+
+// SetConfigFromFile reads and applies blockio configuration from the
+// filesystem.
+func SetConfigFromFile(filename string, force bool) error {
+	if data, err := ioutil.ReadFile(filename); err == nil {
+		if err = SetConfigFromData(data, force); err != nil {
+			return fmt.Errorf("failed to set configuration from file %q: %s", filename, err)
+		}
+		return nil
+	} else {
+		return fmt.Errorf("failed to read config file %q: %v", filename, err)
+	}
+}
+
+// SetConfigFromData parses and applies configuration from data.
+func SetConfigFromData(data []byte, force bool) error {
+	config := &Config{}
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return err
+	}
+	return SetConfig(config, force)
+}
+
+// SetConfig scans available block devices and applies new configuration
```

Comment:
super nit: missing punctuation from the end of the sentence (just to be in line with the other "docstrings")

Reply:
Fixed!